qcore/nhm.py

```diff
@@ -0,0 +1,119 @@
+from typing import List
+
+import numpy as np
+
+
+class NHMInfo:
+    """Contains the information for a single fault from an nhm file
+
+    Attributes
+    ----------
+    fault_name : str
+    tectonic_type : str
+    fault_type : str
+    length_mean : float, (km)
+    length_sigma : float (km)
+    dip_mean : float  (deg)
+    dip_sigma : float (deg)
+    dip_dir : float
+    rup_depth_mean : float (km)
+    rup_depth_sigma : float (km)
+    slip_rate_mean : float (mm/yr)
+    slip_rate_sigma : float (mm/yr)
+    coupling_coeff : float (mm/yr)
+    coupling_coeff_sigma : float (mm/yr)
+    mw_median : float
+    recur_int_median : float (yr)
+    n_locations : int
+        Number of locations on fault surface
+    locations: np.ndarray
+        The fault surface locations, shape [n_locations, 2], format (lon, lat)
+    """
+
+    def __init__(self):
+        """Do not use the constructor, instances should only be
+        created via the from_nhm_section method"""
+        pass
+
+    @classmethod
+    def from_nhm_section(cls, rows: List[str]):
+        """Creates an NHMInfo instance from the given nhm file section
+
+        Parameters
+        ----------
+        rows : List of str
+            The rows of an nhm file corresponding to one specific fault.
+            Format:
+                Row 1: FaultName
+                Row 2: TectonicType , FaultType
+                Row 3: LengthMean , LengthSigma (km)
+                Row 4: DipMean , DipSigma (deg)
+                Row 5: DipDir
+                Row 6: Rake (deg)
+                Row 7: RupDepthMean , RupDepthSigma (km)
+                Row 8: RupTopMean, RupTopMin RupTopMax  (km)
+                Row 9: SlipRateMean , SlipRateSigma (mm/yr)
+                Row 10: CouplingCoeff , CouplingCoeffSigma (mm/yr)
+                Row 11: MwMedian , RecurIntMedian  (yr)
+                Row 12: Num Locations on Fault Surface
+                Row 13+: Location Coordinates (Long, Lat)
+
+        Returns
+        -------
+        NHMInfo
+        """
+        # Not sure of a tidier of doing this (without adding all these to the init..)
+        info = NHMInfo()
```

Comment:
Just curious. Can't you put all these in the init, and then, return NHMInfo(x,x,x,x,x,)?